handbook/engineering/search/index.md

```diff
@@ -2,9 +2,94 @@
 
 We own the end-to-end Sourcegraph search experience.
 
+## Vision
+
+Sourcegraph search is:
+
+- **Fast:** Performance of showing results is fast and I can get to what I’m looking for quickly.
+- **Scalable:** Sourcegraph scales to any private codebase and the size of the open-source universe. Users can trust that the result set is drawn from the entire set of code they care about, and if the result isn't found, it doesn't exist.
+- **Expressive:** It is possible to construct a search query to find exactly the results I’m looking for. The tools, syntax, filters, are provided to construct the queries I want to write. We strive for a search experience where the answer to “Can I do ‘X’ with Sourcegraph?” will always be yes.
+- **Easy to use:** Users can quickly understand how to find what they are looking for and what options are available for searching. The search syntax is clear and intuitive.
+- **Shareable:** Sourcegraph searches are easy to share, and provide team and organization wide value. This in turn creates network effects that compound the value Sourcegraph provides with scale.
+
 ## Goals
 
-Our current goals are documented in the [search team's current tracking issue](https://github.com/sourcegraph/sourcegraph/issues?q=is%3Aopen+is%3Aissue+label%3Atracking+label%3Ateam%2Fsearch).
+_Updated 2020-08-27_
+
+## Long term (~1 year)
+
+Sourcegraph search is far and away the best code search solution, search results are fast, and I can find what I’m looking for quickly.
+
+## Medium term (3-6 months)
+
+In the next 6 months we want to be halfway to the long term goal. To achieve this, we need to accomplish the following outcomes:
+
+- Getting to search results faster through a better search experience.
+- Users perceive search results to be as fast-loading and responsive as [grep.app](https://grep.app/)
+- Scale fast search to 500k repositories.
+
+## Short term (1-3 months)
+
+Our current focus is documented in the [tracking issue for the current milestone](https://github.com/sourcegraph/sourcegraph/issues?q=is%3Aopen+is%3Aissue+label%3Atracking+label%3Ateam%2Fsearch).
+
+### Fast
+
+- **Unblock improving user perceived performance**
+  - **Problem:** We want to make significant improvements to search that are blocked by not having the infrastructure needed.
```

Comment:
This doesn't feel like a problem statement.

What about this?

```suggestion
  - **Problem:** Our current search infrastructure relies on on-shot requests with aggressive timeouts, but it is hard to tune those timeouts for large codebases. If we make the timeouts too small, then search may not return results for needle-in-the-haystack queries (because the search timed out). If we make the timeouts too large, then search becomes slower across the board. Detecting and handling these timeouts across multiple service boundaries is also error prone and has led to bugs.
```

Is this accurate, @keegancsmith ? If not, what specific problem does streaming search solve?

Reply:
Sounds good. I would also emphasize the larger the timeout, the longer a user will wait to receive a small result set. Main benefit is a user will start getting results from the very first result we find, rather than waiting for the full one-shot search (which can be multiple seconds with certain queries on large instances)